exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/DrillFilterAggregateTransposeRule.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.planner.logical;
+
+import org.apache.calcite.rel.rules.FilterAggregateTransposeRule;
+
+public class DrillFilterAggregateTransposeRule{
```

Comment:
@jacques-n @amansinha100 , could you please take another look at this PR? I modified the patch based on the discussion we had.

In stead of creating two instances (which causes Calcite complain duplicate rule descriptions error), I override match() method to check the rels have the same convention trait.

As a separate work, I'm working on DRILL-3855, to re-enable two rules (FilterSetOpTranspose, ProjectSetOpTranspose). These rules were disabled in DRILL-3257, since it caused TPC-DS Q74 to hang in planning, very similar to what was reported in this JIRA.